### DISCUSSION TITLE
column wise filter

### DISCUSSION THREAD

**Discussion by sunilit42** [rappasoft/laravel-livewire-tables#388] [Feature Requests]
Hello,

your module works fine but i want column wise search functionlity

e.g customer email , i want to search inside specific column

i see one of plugin provide that https://github.com/Power-Components/livewire-powergrid

i already implement lot of grid using your plugin so i don't want to change it

hope you will provide some solution on that

**Comment by rappasoft** [comment]
We don't have column search functionality. But you can control the search query on a per column basis.

**Reply by sunilit42** [reply]
without filter add what is use of search query on a per column basis

=> i did below thing

 <input
                        data-id="customer_email"
                        wire:model="filters.customer_email"  />
                        
                        
added into filter filter

In filter.php file

public const TYPE_TEXT = 'text';

public function isText(): bool
    {
        return $this->type === self::TYPE_TEXT;
    }
    
  WithFilters file
   added below line no after 170
            if ($filterDefinitions[$filterName]->isText()) {
                return  true;
            }
            
and into table file
return Orders::select('orders.*', 'user_stores.name as storename')
            ->join('user_stores', 'user_stores.id', '=', 'orders.store')
            ->whereIn('orders.store', $stores)
            ->when($this->getFilter('store'), function ($query, $storeId) {
                if ($storeId > 0) {
                    return $query->where('user_stores.id',$storeId);
                }
            })->when($this->getFilter('customer_email'), function ($query, $value) {
               return $query->where('customer_email', 'like', '%' . $value . '%');
            });

Hope i am right

only issue with that it is calling on change but i want to on key up

**Reply by rappasoft** [reply]
You should not change the core files because the next time you run composer I could have updated them and they will be replaced. If you extended them you run in to the same issue of me changing the core. But if it's working for you then just lock the version in composer.

**Reply by sunilit42** [reply]
yes i know i m extending your plugin but just asking your my code is ok ?

how to do call on keyup not on change?
